.github/workflows/build.yml

```diff
@@ -20,27 +20,145 @@ on:
       - '!gh-pages'
 
 jobs:
-  build:
+  # verify build on one node before multiple builds on different os are started
+  fail-fast-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        id: java8
+        uses: actions/setup-java@v2
+        with:
+          distribution: temurin
+          java-version: '8'
+
+      - name: Set up JDK 11
+        id: java11
+        uses: actions/setup-java@v2
+        with:
+          distribution: temurin
+          java-version: '11'
+          cache: 'maven'
+
+      - name: Setup maven toolchains.xml
+        shell: bash
+        run: |
+          echo "JAVA8_HOME=${{ steps.java8.outputs.path }}" >> $GITHUB_ENV
+          echo "JAVA11_HOME=${{ steps.java11.outputs.path }}" >> $GITHUB_ENV
+          cp tools/toolchains.xml $HOME/.m2/
+
+      - name: Build with Maven
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: bash tools/build.sh -Dtarget.platform=oxygen
+
+  verify:
+    needs: fail-fast-build
 
     strategy:
       matrix:
         os: [ ubuntu-latest, windows-latest, macOS-latest ]
-      fail-fast: false
+        target: [ oxygen, 2022-06 ]
+        exclude:
+          # exclude the fail-fast-build, which already run
+          - os: ubuntu-latest
+            target: oxygen
+      fail-fast: true
 
     runs-on: ${{ matrix.os }}
 
-    if: ${{ !contains(github.event.head_commit.message, '[skip ci]') }}
```

Comment:
Note: this works now out of the box in github actions: https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/